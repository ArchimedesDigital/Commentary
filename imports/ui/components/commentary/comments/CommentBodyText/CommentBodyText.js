
/*
	helpers
*/

const createRevisionMarkup = (html) => {
	let newHtml = html;

	const workNamesSpace = [{
		title: 'Iliad',
		slug: 'iliad',
	}, {
		title: 'Odyssey',
		slug: 'odyssey',
	}, {
		title: 'Homeric Hymns',
		slug: 'hymns',
	}, {
		title: 'Hymns',
		slug: 'hymns',
	}];
	const workNamesPeriod = [{
		title: 'Il',
		slug: 'iliad',
	}, {
		title: 'Od',
		slug: 'odyssey',
	}, {
		title: 'HH',
		slug: 'hymns',
	}, {
		title: 'I',
		slug: 'iliad',
	}, {
		title: 'O',
		slug: 'odyssey',
	}];

	let regex1;
	let regex2;

	workNamesSpace.forEach((workName) => {
		// regex for range with dash (lookahead to ignore if surrounded by &quot; - required for comment cross reference)
		regex1 = new RegExp(`${workName.title} (\\d+).(\\d+)-(\\d+)(?!.*&quot;)`, 'g');

		// regex for no range (and lookahead to ensure range isn't captured) (lookahead to ignore if surrounded by &quot; - required for comment cross reference)
		regex2 = new RegExp(`${workName.title} (\\d+).(?!\\d+-\\d+)(\\d+)(?!.*&quot;)`, 'g');

		newHtml = newHtml.replace(regex1,
			`<a
				class='has-lemma-reference'
				data-work=${workName.slug}
				data-subwork='$1'
				data-lineFrom='$2'
				data-lineTo='$3'
			>${workName.title} $1.$2-$3</a>`);
		newHtml = newHtml.replace(regex2,
			`<a
				class='has-lemma-reference'
				data-work=${workName.slug}
				data-subwork='$1'
				data-lineFrom='$2'
			>${workName.title} $1.$2</a>`);
	});

	workNamesPeriod.forEach((workName) => {
		// regex for range with dash (lookahead to ignore if surrounded by &quot; - required for comment cross reference)
		regex1 = new RegExp(`([^\\w+])${workName.title}.(\\s*)(\\d+).(\\d+)-(\\d+)(?!.*&quot;)`, 'g');

		// regex for no range (and lookahead to ensure range isn't captured) (lookahead to ignore if surrounded by &quot; - required for comment cross reference)
		regex2 = new RegExp(`([^\\w+])${workName.title}.(\\s*)(\\d+).(?!\\d+-\\d+)(\\d+)(?!.*&quot;)`, 'g');
		newHtml = newHtml.replace(regex1,
			`$1<a
				class='has-lemma-reference'
				data-work=${workName.slug}
				data-subwork='$3'
				data-lineFrom='$4'
				data-lineTo='$5'
			>${workName.title}.$2$3.$4-$5</a>`);
		newHtml = newHtml.replace(regex2,
			`$1<a
				class='has-lemma-reference'
				data-work=${workName.slug}
				data-subwork='$3'
				data-lineFrom='$4'
			>${workName.title}.$2$3.$4</a>`);
	});

	return { __html: newHtml };
};

/*
	BEGIN CommentBodyText
*/
const CommentBodyText = props => (
	<div
		className="comment-body"
		dangerouslySetInnerHTML={props.createRevisionMarkup ? createRevisionMarkup(props.text) : { __html: props.text }}
		onClick={props.onTextClick}
	/>
);
CommentBodyText.propTypes = {
	text: React.PropTypes.string.isRequired,
	onTextClick: React.PropTypes.func,
	createRevisionMarkup: React.PropTypes.bool,
};
CommentBodyText.defaultProps = {
	onTextClick: null,
	createRevisionMarkup: false,
};
/*
	END CommentBodyText
*/

export default CommentBodyText;
