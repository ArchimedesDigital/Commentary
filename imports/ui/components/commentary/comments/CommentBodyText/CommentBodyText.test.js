import React from 'react';
import renderer from 'react-test-renderer';

// component:
import CommentBodyText from './CommentBodyText';

describe('CommentBodyText', () => {
	it('renders correctly', () => {

		const tree = renderer
			.create(<CommentBodyText text="" />)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
