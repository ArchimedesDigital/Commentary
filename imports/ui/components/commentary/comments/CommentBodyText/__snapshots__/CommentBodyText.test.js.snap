// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentBodyText renders correctly 1`] = `
<div
  className="comment-body"
  dangerouslySetInnerHTML={
    Object {
      "__html": "",
    }
  }
  onClick={null}
/>
`;
